service/history/workflow/state_transition_history.go

```diff
@@ -108,23 +108,25 @@ func TransitionHistoryStalenessCheck(
 // CompareVersionedTransition compares two VersionedTransition structs.
 // Returns -1 if a < b, 0 if a == b, 1 if a > b.
 //
-// A VersionedTransition  is considered less than another
+// A VersionedTransition is considered less than another
 // if its NamespaceFailoverVersion is less than the other's.
 // Or if the NamespaceFailoverVersion is the same, then the TransitionCount is compared.
+// Nil is considered the same as empty versioned transition with both fields set to 0,
+// thus smaller than any non-nil versioned transition.
```

Comment:
I'd prefer if we had a sentinel zero value for this so the behavior is more intuitive when reading the code and to avoid NPEs in cases the code makes assumptions that a versioned transition isn't nullable.